source/common/http/utility.cc

```diff
@@ -93,26 +93,36 @@ std::string Utility::createSslRedirectPath(const HeaderMap& headers) {
 }
 
 Utility::QueryParams Utility::parseQueryString(absl::string_view url) {
-  QueryParams params;
   size_t start = url.find('?');
   if (start == std::string::npos) {
+    QueryParams params;
     return params;
   }
 
   start++;
-  while (start < url.size()) {
-    size_t end = url.find('&', start);
+  return parseParameters(url, start);
+}
+
+Utility::QueryParams Utility::parseFormBody(absl::string_view body) {
+  return parseParameters(body, 0);
+}
+
+Utility::QueryParams Utility::parseParameters(absl::string_view data, size_t start) {
+  QueryParams params;
+
+  while (start < data.size()) {
```

Comment:
Can you use  StringUtil::SplitToken() (https://github.com/envoyproxy/envoy/blob/fa0b52a874b67c857bd314620592d39280767330/source/common/common/utility.h#L280) to split the params on `&`?

...oh I see the code already did this, and it uses subspan() which return a string rather than a string_view so it has to do a bunch of memory allocation and copying...OK but I guess that's current condition. Can you just leave a TODO to improve?

Reply:
I am happy to port this to `SplitToken()`, to avoid the allocations + copies.

Reply:
hmm actually we'd still need the allocation to add the key/value via `emplace()` to convert the string_views to std::strings, which would be same amount of allocation+copying with subspan + calling `emplace(&&.., &&... )`, no?

Reply:
ok, makes sense.